onadata/apps/api/viewsets/note_viewset.py

```diff
@@ -57,12 +61,35 @@ class NoteViewSet(ViewPermissionMixin, ModelViewSet):
   >
   >        HTTP 200 OK
 """
-    queryset = Note.objects.all()
     serializer_class = NoteSerializer
-    permission_classes = [permissions.ViewDjangoObjectPermissions,
-                          permissions.IsAuthenticated, ]
-
-    #u This used to be post_save. Part of it is here, permissions validation
+    permission_classes = [NoteObjectPermissions]
+
+    def get_queryset(self):
```

Comment:
I think you could replace a lot of this work with a call to https://django-guardian.readthedocs.io/en/v1.4.9/api/guardian.shortcuts.html#get-objects-for-user. You'd still have to add in the `shared_data` forms; it might look something like:
```python
viewable_xforms = get_objects_for_user(…)
viewable_notes = Note.objects.filter(
    Q(instance__xform=viewable_xforms) | Q(instance__xform__shared_data=True)
)
```

Reply:
> 😁 When do we start adding unit tests to all new KC work? 😁 I guess after KoBoKitten is done?

I don't think we have to wait for KoBoKitten to be ready. If new tests work in the current broken version, they will in KoBoKitten.